mev-build-rs/src/auctioneer.rs

```diff
@@ -50,7 +51,25 @@ fn prepare_submission(
     };
     let execution_payload = to_execution_payload(payload.block());
     let signature = sign_builder_message(&message, signing_key, context)?;
-    Ok(SignedBidSubmission { message, execution_payload, signature })
+    let submission = match execution_payload.version() {
+        Fork::Bellatrix => {
+            SignedBidSubmission::Bellatrix(block_submission::bellatrix::SignedBidSubmission {
+                message,
+                execution_payload,
+                signature,
+            })
+        }
+        Fork::Capella => {
+            SignedBidSubmission::Capella(block_submission::capella::SignedBidSubmission {
+                message,
+                execution_payload,
+                signature,
+            })
+        }
+        Fork::Deneb => unimplemented!(),
```

Comment:
`payload` fn arg has a method to get blob sidecars, which would need to be mapped into the type here

may also be a good opportunity for a conversion function near where `SignedBidSubmission` is defined

Reply:
I don't see that [`EthBuiltPayload`](https://github.com/paradigmxyz/reth/blob/ffa36b7348b2e5f4bdb09804961eda597131b91d/crates/ethereum/engine-primitives/src/payload.rs#L26) has a method to retrieve the blob sidecars. it looks like we may need to go through [`ExecutionPayloadEnvelopeV3`](https://github.com/paradigmxyz/reth/blob/ffa36b7348b2e5f4bdb09804961eda597131b91d/crates/ethereum/engine-primitives/src/payload.rs#L106).

Reply:
yeah, I would just focus on the boost and relay stuff first, this is a bit annoying and im just gonna try to expose that field in reth

Reply:
I opened a PR [here](https://github.com/paradigmxyz/reth/pull/7979).